Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift

```diff
@@ -853,6 +853,7 @@ private final class TokenStreamCreator: SyntaxVisitor {
   }
 
   func visit(_ node: FunctionParameterSyntax) -> SyntaxVisitorContinueKind {
+    after(node.attributes?.lastToken, tokens: .break)
```

Comment:
Thinking about this further after my initial Twitter comment, I realize this won't handle stacked property wrappers—if you add a test like this, I bet it would mash them together:

```swift
func foo<Content: View>(@Foo @ViewBuilder bar: () -> View) {
  bar()
}
```

So instead of inserting this break, can you replace this line with a call to `arrangeAttributeList(node.attributes)`? That will do all the work of inserting breaks between the individual attributes, after the last one, and also adding some groups to keep them together when possible. You can see an example of this on line 939-ish in the `visit(EnumCaseDeclSyntax)` method.

(The reason this has to be an explicit call instead of happening automatically as part of the regular visitation of the syntax tree is due to the visitation order w.r.t. the way the before/after maps are built up.)

Reply:
great call! just developed against the 5.1 branch and the stacked case was indeed broken. your suggested fix works like a charm!